gamemode/modules/base/sv_gamemode_functions.lua

```diff
@@ -312,29 +313,38 @@ for _, ply in pairs(player.GetHumans()) do
     DrpCanHear[ply] = {}
 end
 
-local gridSize = GM.Config.voiceDistance*2 -- Grid cell size is equal to the size of the diamater of player talking
+local gridSize = GM.Config.voiceDistance * 2 -- Grid cell size is equal to the size of the diamater of player talking
 local floor = math.floor -- Caching floor as we will need to use it a lot
 
 -- Grid based position check
-timer.Create("DarkRPCanHearPlayersVoice", 1, 0, function()
-    DrpCanHear = {}
+local grid
+local plyToGrid = {
+    {},
+    {}
+}
+
+
+timer.Create("DarkRPCanHearPlayersVoice", timeDelay, 0, function()
     local players = player.GetHumans()
 
-    -- Create Grid
-    local grid = {}
-    local plyToGrid = {
-        {},
-        {}
-    }
+
+    -- Clear old values
+    plyToGrid[0] = {}
+    plyToGrid[1] = {}
+    grid = {}
 
     -- Get the grid position of every player O(N)
     for _, ply in ipairs(players) do
-        local pos = ply:GetShootPos()
-        local x = floor(pos.x/gridSize)
-        local y = floor(pos.y/gridSize)
-        grid[x] = grid[x] or {}
-        grid[x][y] = grid[x][y] or {}
-        grid[x][y][#grid[x][y] + 1] = ply
+        local pos = ply:GetPos()
+        local x = floor(pos.x / gridSize)
+        local y = floor(pos.y / gridSize)
+
+        local row = grid[x] or {}
+            local cell = row[y] or {}
+                table.insert(cell, ply)
+            row[y] = cell
+        grid[x] = row
```

Comment:
Why is there weird indentation?

Reply:
I personally find it easier to read, don't know how others feel about it.

Reply:
I think that sort of indentation should only be used for scopes and stack operations, it looks weird in this context.